packages/snaps-jest/src/internals/simulation/store/mocks.ts

```diff
@@ -0,0 +1,55 @@
+import type { Json } from '@metamask/utils';
+import type { PayloadAction } from '@reduxjs/toolkit';
+import { createSelector, createSlice } from '@reduxjs/toolkit';
+
+import type { ApplicationState } from './store';
+
+export type JsonRpcMock = {
+  method: string;
+  result: Json;
+};
+
+export type MocksState = {
+  jsonRpc: Record<string, Json>;
+};
+
+/**
+ * The initial notifications state.
+ */
+const INITIAL_STATE: MocksState = {
+  jsonRpc: {},
+};
+
+export const mocksSlice = createSlice({
+  name: 'mocks',
+  initialState: INITIAL_STATE,
+  reducers: {
+    addJsonRpcMock: (state, action: PayloadAction<JsonRpcMock>) => {
+      // @ts-expect-error - TS2589: Type instantiation is excessively deep and
+      // possibly infinite.
+      state.jsonRpc[action.payload.method] = action.payload.result;
```

Comment:
Out of scope but we should probably allow multiple mocks for the same method or specifying a function

Reply:
Maybe create a ticket for this?

Reply:
#2096.